<?php
$page_overview="<p>
The new field of metagenomics studies the microorganism communities by culture-independent sequencing. With the advances in the next-generation sequencing techniques, scientists are facing tremendous challenges in metagenomic data analysis due to the huge quantity and high complexity of sequence data. Analyzing large datasets is extremely time-consuming. In addition, metagenomic annotation involves a wide range of computational tools. It is very difficult for common users to install and maintain these software packages. The tools provided by the few available web servers are also limited and have various constraints such as login requirement, long waiting time, inability to configure pipelines etc. Here we developed WebMGA, a customizable web server for fast metagenomic analysis. WebMGA includes over 20 commonly used tools for analyses such as ORF calling, sequence clustering, quality control of raw reads, removal of sequencing artifacts and contaminations, taxonomic analysis, functional annotation etc, as shown in the following diagram:
</p>
<img class=\"centered\" src=\"image/webmga_figs.002s650.png\" />
<p>
WebMGA provides users with rapid metagenomic data analysis using fast and effective algorithms. All the tools behind WebMGA were implemented to run in parallel on our local computer cluster. Users can access WebMGA through web browsers or programming scripts to perform individual analysis or to configure and run customized pipelines. WebMGA is freely available at <a href=\"http://weizhongli-lab.org/webMGA\">http://weizhongli-lab.org/webMGA</a>
</p>
<p>
The server backend was developed by Sitao Wu. And the frontend was developed by Zhengwei Zhu.
</p>
";
?>
